rlog/2024-10-22-gsub-idontwant-tests.mdx

```diff
@@ -0,0 +1,110 @@
+<!--truncate-->
+
+## Overview
+
+[IDONTWANT](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.2.md?plain=1#L52) messages are introduced to curtail redundant transmissions without compromising resilience. 
+Cutting down on duplicates can potentially render two significant advantages
+
+1. Minimize bandwidth requirements 
+2. Minimize message dissemination time
+
+For IDONTWANTs to be effective, they must be received and processed by the sender before it starts relaying the message. 
+[Duplicates investigation](https://ethresear.ch/t/number-duplicate-messages-in-ethereums-gossipsub-network/19921#arrival-time-of-duplicates-9) reveals that 
+the average time difference between the first message arrival and the first duplicate arrival is higher than the average round trip time in Ethereum's GossipSub network. 
+
+This allows for timely IDONTWANT reception and canceling of many duplicate transmissions, 
+showing a potential for a significant drop in bandwidth utilization. 
+On the other hand, lowering message dissemination time is only possible by minimizing queuing delays at busy peers. 
+
+## Experiments
+
+We conducted a series of experiments with different arrangements (changing heartbeat_interval and message size) 
+to precisely identify the impact of IDONTWANT messages on bandwidth utilization and message dissemination time. 
+
+The experiments are performed on nim-libp2p using the [shadow simulator](https://github.com/vacp2p/dst-gossipsub-test-node/pull/4). 
+The peer bandwidth and link latency are uniformly set between 50-150 Mbps and 40-130 milliseconds in five stages. 
+
+Every publisher transmits exactly one message to the network, 
+and inter-packet spacing (delay) is set to four seconds. 
+In all experiments, ten messages are transmitted to the network. 
+
+At the start of each experiment, two additional messages are transmitted to raise the $C_{wnd}$ to a reasonable level. 
+These messages are not included in latency computations. The simulation details are presented in the table below.
+
+
+| **Parameter** | **Value** | **Parameter** | **Value** |
+| ------------- | --------- | ------------- | --------- |
+| Peers | 2000 | Publishers | 10 |
+| Peer Bandwidth | 50-150 Mbps | Link Latency | 40-130 ms |
+| Message Size | 1KB, 50KB, 500KB, 1MB | $D$ | 8 |
+| Heartbeat Interval | 700ms, 1000ms, 1500ms | $D_{low}$ | 6 |
+| FloodPublish | False | $D_{high}$ | 12 |
+| Gossip factor | 0.05 | Muxer | yamux |
+
+
+## Findings
+
+Experiments reveal that IDONTWANT messages yield a noticeable (up to 21%) drop in bandwidth utilization. 
+A higher drop is seen with a higher heartbeat interval. 
+Interestingly, a relatively low bandwidth reduction (12-20%) is seen for 1MB messages, 
+compared to 500KB messages (18-21%).   
+
+
+| ![Image 1](/img/idontwanttest/BW_700ms.png) | ![Image 2](/img/idontwanttest/BW_1000ms.png) | ![Image 3](/img/idontwanttest/BW_1500ms.png) |

```

Comment:
The bandwidth represented here is the sum of the total bandwidth of the simulation, right? If I am correct, I think it should be nice to clarify this, because at first it is a bit confusing (to me at least)